src/mlpack/methods/preprocess/preprocess_scale_main.cpp

```diff
@@ -0,0 +1,174 @@
+    "standard scaler, and outputs a matrice with scaled feature."
+    "\n\n"
+    "The output scaled feature matrices may be saved with the " +
+    PRINT_PARAM_STRING("output") + " output parameters."
+    "\n\n"
+    "The model to scale features can be saved using " +
+    PRINT_PARAM_STRING("output_model") + " and later can be loaded back using"
+    + PRINT_PARAM_STRING("input_model") + "."
+    "\n\n"
+    "So, a simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled")+ " with "
+    " standard_scaler as scaler_method, we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "standard_scaler") +
+    "\n\n"
+    "A simple example where we want to whiten the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_whitened")+ " with "
+    " PCA as whitening_method and use 0.01 as regularization parameter, "
+    "we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "pcawhitening", "epsilon", 0.01) +
+    "\n\n"
+    "You can also retransform the scaled dataset back using" +
+    PRINT_PARAM_STRING("function") + ". An example to rescale : " +
+    PRINT_DATASET("X_scaled") + " into " + PRINT_DATASET("X")
+    + "using the saved model " + PRINT_PARAM_STRING("input_model") + " is:"
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X_scaled", "output", "X",
+    "function", 1, "input_model", "saved")+
+    "\n\n"
+    "Another simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled") + " with "
+    " min_max_scaler as scaler method, where scaling range is 1 to 3 instead"
+    " of deafult 0 to 1. We coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "min_max_scaler", "min_value", 1, "max_value", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save scaled data to.", "o");
+PARAM_STRING_IN("scaler_method", "method to use for scaling, the "
+    "default is standard_scaler.", "a", "standard_scaler");
+PARAM_DOUBLE_IN("epsilon", "regularization Parameter for pcawhitening,"
+    " or zcawhitening, should be between -1 to 1.", "r", 0.000001);
+
+PARAM_INT_IN("seed", "Random seed (0 for std::time(NULL)).", "s", 0);
+PARAM_INT_IN("min_value", "Starting value of range for min_max_scaler.",
+    "b", 0);
+PARAM_INT_IN("max_value", "Ending value of range for min_max_scaler.",
+    "e", 1);
+PARAM_INT_IN("function", "function to apply,either 0:Transform or 1:Inverse"
+  "Transform", "f", 0)
```

Comment:
Consider using `PARAM_FLAG_IN()` and maybe calling it `"inverse_scaling"`, so it will apply regular scaling if not specified but inverse scaling if it is.

Reply:
I guess you are talking about `PARAM_FLAG()` , because i didn't find `PARAM_FLAG_IN()` anywhere